client/src/client/workers/test-evaluator.ts

```diff
@@ -111,6 +104,10 @@ ctx.onmessage = async (e: TestEvaluatorEvent) => {
 
   const assert = chai.assert;
   const __helpers = helpers;
+  // If the learner tries to modify these, weird behavior may result. Freezing
+  // them means they get an error instead.
+  Object.freeze(assert);
+  Object.freeze(__helpers);
```

Comment:
Why do we not also freeze `__utils`?

Reply:
It's frozen, but at the end of the file. I'll move it up so it's more obvious.

Reply:
🤦‍♂️